linebot/api.py

```diff
@@ -124,6 +124,221 @@ def push_message(self, to, messages, timeout=None):
+
+        return RichMenu.new_from_json_dict(response.json)
+
+    def create_rich_menu(self, data, timeout=None):
+        """Call delete rich menu API.
+
+        https://developers.line.me/en/docs/messaging-api/reference/#create-rich-menu
+
+        Create a rich menu object through a group of given data
+
+        :param data: Inquired to create a rich menu.
+        :type data: T <= :py:class:`linebot.models.template.RichMenuTemplate`
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        :return: An RichMenu instance with a rich menu id
+        :type RichMenu: T <= :py:class:`linebot.models.reponse.RichMenu`
+        """
+        response = self._post(
+            '/v2/bot/richmenu', data=data.as_json_string(), timeout=timeout
+        )
+
+        return RichMenu.new_from_json_dict(response.json)
+
+    def link_rich_menu_to_user(self, user_id, rich_menu_id, timeout=None):
+        """Call link rich menu to user API.
+
+        https://developers.line.me/en/docs/messaging-api/reference/#link-rich-menu-to-user
+
+        Links a rich menu to a user. Only one rich menu can be linked to a user at one time.
+
+        :param str user_id: ID of an uploaded rich menu
+        :param str rich_menu_id: ID of the user
+        :type timeout: float | tuple(float, float)
+        :return: An empty RichMenu instance
+        :type RichMenu: T <= :py:class:`linebot.models.reponse.RichMenu`
+        """
+        response = self._post(
+            '/v2/bot/user/{user_id}/richmenu/{rich_menu_id}'.format(
+                user_id=user_id,
+                rich_menu_id=rich_menu_id
+            ),
+            timeout=timeout
+        )
+
+        return RichMenu.new_from_json_dict(response.json)
+
+    def get_rich_menu_list(self, timeout=None):
+        """Call get rich menu list API.
+
+        https://developers.line.me/en/docs/messaging-api/reference/#get-rich-menu-list
+
+        Gets a list of all uploaded rich menus.
+
+        :return: An list of RichMenu instances
+        :type RichMenu: T <= :py:class:`linebot.models.reponse.RichMenu`
+        :param timeout: (optional) How long to wait for the server
+            to send data before giving up, as a float,
+            or a (connect timeout, read timeout) float tuple.
+            Default is self.http_client.timeout
+        :type timeout: float | tuple(float, float)
+        """
+        response = self._get(
+            '/v2/bot/richmenu/list',
+            timeout=timeout
+        )
+
+        lst_result = []
+        for richmenu in response.json['richmenus']:
+            lst_result.append(RichMenu.new_from_json_dict(richmenu))
+
+        return lst_result
+
+    def upload_rich_menu_image(self, rich_menu_id, path_to_image, timeout=None):
```

Comment:
Could you change to `set_rich_menu_image` instead of `upload_rich_menu_image` ?
We would like to unify the names with other SDKs.

And I think it is better to pass binary and content-type.
`def set_rich_menu_image(self, rich_menu_id, content_type, content, timeout=None)`

e.g.
https://github.com/line/line-bot-sdk-nodejs/blob/master/lib/client.ts#L157
https://github.com/line/line-bot-sdk-java/blob/master/line-bot-api-client/src/main/java/com/linecorp/bot/client/LineMessagingClient.java#L193